tracer/src/Datadog.Trace/OTelMetrics/MetricPoint.cs

```diff
@@ -84,7 +112,58 @@ private static int FindBucketIndex(double value)
             }
         }
 
-        return DefaultHistogramBounds.Length;
+        return DefaultHistogramBounds.Length; // Overflow bucket
+    }
+
+    /// <summary>
+    /// Creates a snapshot copy of the current metric state for export.
+    /// For delta temporality, this resets the running values after taking the snapshot.
+    /// For cumulative temporality, this preserves the running values.
+    /// The temporality behavior is determined by the AggregationTemporality set at construction.
+    /// </summary>
+    /// <returns>A new MetricPoint containing the snapshot values</returns>
+    public MetricPoint CreateSnapshotAndReset()
+    {
+        lock (_histogramLock)
+        {
+            var endTime = DateTimeOffset.UtcNow;
+
+            // Create a snapshot copy with current values
+            var snapshot = new MetricPoint(InstrumentName, MeterName, InstrumentType, AggregationTemporality, Tags, Unit, Description)
+            {
+                StartTime = this.StartTime,
+                EndTime = endTime,
+                SnapshotCount = _runningCountValue,
+                SnapshotSum = _runningDoubleValue,
+                SnapshotGaugeValue = _runningDoubleValue,
+                SnapshotMin = _runningMin,
+                SnapshotMax = _runningMax
+            };
+
+            // Copy bucket counts for histograms
+            if (_runningBucketCounts.Length > 0)
+            {
+                snapshot.SnapshotBucketCounts = new long[_runningBucketCounts.Length];
+                Array.Copy(_runningBucketCounts, snapshot.SnapshotBucketCounts, _runningBucketCounts.Length);
+            }
+
+            if (AggregationTemporality == OTelMetrics.AggregationTemporality.Delta)
+            {
+                _runningCountValue = 0;
+                _runningDoubleValue = 0.0;
+                _runningMin = double.PositiveInfinity;
+                _runningMax = double.NegativeInfinity;
+                if (_runningBucketCounts.Length > 0)
+                {
+                    Array.Clear(_runningBucketCounts, 0, _runningBucketCounts.Length);
+                }
+
+                // Update start time for next delta window
+                StartTime = endTime;
```

Comment:
I'm going to follow up on this, but I don't know if we are supposed to update StartTime each time we collect...I think we only need to update EndTime. I'll verify this offline and come back to this comment

Reply:
Looking at this it seems like we can leave it as it is, WDYT?
- https://buf.build/opentelemetry/opentelemetry/docs/main:opentelemetry.proto.metrics.v1#:~:text=rejected%20by%20consumers.-,StartTimeUnixNano,-StartTimeUnixNano%20in%20general

Reply:
Ohhh I see, in the Python OTel SDK implementation, this gets updated each time (what you're doing!) for Delta aggregations. But for Cumulative, it's not updated and it's just the original time. https://github.com/open-telemetry/opentelemetry-python/blob/daae23bc7f5514003d1e262730c409875d12d77d/opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/aggregation.py#L333-L366

So this change looks good! Feel free to resolve this thread.